### PR TITLE
Show Global Events when clicking on them in the log.

### PR DESCRIPTION
**Pull request by kberg**
image

### CODE REVIEW

**Comment by kberg** [comment]
Fixes #3057

**Comment by kevinb9n** [comment]
random: the new boxed card name style is nice. If it helps, the physical cards have a background color of approx. #a8d8e8 and I think that color would work well here.

File: src/components/LogPanel.ts [hunk 1]
```diff
@@ -267,20 +313,23 @@ export const LogPanel = Vue.component('log-panel', {
               {{ n }}
             </div>
           </div>
-          <span class="label-additional" v-html="getGenerationText()"></span>
+          <span class="label-additional" v-if="players.length === 1"><span :class="lastGenerationClass">of {{this.lastSoloGeneration}}</span></span>
```
Comment (bafolts) [inline src/components/LogPanel.ts:316]: Winner! Every `v-html` removed is a step in right direction.

File: src/turmoil/Turmoil.ts [hunk 2]
```diff
@@ -233,9 +233,11 @@ export class Turmoil implements ISerializable<SerializedTurmoil> {
 
       // 2 - Global Event
       // TODO: create a LogMessageDataType for Global Events, ref https://github.com/bafolts/terraforming-mars/issues/3057
-      if (this.currentGlobalEvent) {
-        game.log('Global event "' + this.currentGlobalEvent.name + '" has been resolved.');
-        this.currentGlobalEvent.resolve(game, this);
+      if (this.currentGlobalEvent !== undefined) {
+        const currentGlobalEvent: IGlobalEvent = this.currentGlobalEvent;
+        game.log('Resolving global event ${0}', (b) => b.globalEvent(currentGlobalEvent));
+        // game.log('Global event "' + this.currentGlobalEvent.name + '" has been resolved.');
```
Comment (bafolts) [inline src/turmoil/Turmoil.ts:239]: Remove comment before flight.

**Review by bafolts — APPROVED** [review]

**Comment by kberg** [comment]
> random: the new boxed card name style is nice. If it helps, the physical cards have a background color of approx. #a8d8e8 and I think that color would work well here.

Probably true, not the simplest thing for me to do, amusingly. I'll follow up with this.

**Comment by kberg** [comment]
image
OK.

File: src/components/Turmoil.ts [hunk 3]
```diff
@@ -234,26 +235,17 @@ export const Turmoil = Vue.component('turmoil', {
...
-            <div class="event-party event-party--top" :class="'event-party--'+partyNameToCss(turmoil.current.revealed)" v-i18n>{{ turmoil.current.revealed }}</div>
-            <div class="event-party event-party--bottom" :class="'event-party--'+partyNameToCss(turmoil.current.current)" v-i18n>{{ turmoil.current.current }}</div>
-            <div class="event-content" v-i18n>{{ turmoil.current.description }}</div>
-          </div>
+        <global-event v-if="turmoil.distant" :globalEvent="turmoil.distant" type="distant"></global-event>
+        <global-event v-if="turmoil.coming" :globalEvent="turmoil.coming" type="coming"></global-event>
+        <global-event v-if="turmoil.current" :globalEvent="turmoil.current" type="current"></global-event>
       </div>
-
+^
```
Comment (alrusdi) [inline src/components/Turmoil.ts:248]: Is it real symbol or just github glitch?

Comment (kberg) [reply src/components/Turmoil.ts:248]: Dang.

**Review by kberg — COMMENTED** [review]
I'll fix that.

File: src/components/Turmoil.ts (hunk 3 above)
Comment (kberg) [reply src/components/Turmoil.ts]: Fixed, thanks.

**Pull request merged**